File: spring/user/src/main/java/com/admin/user/controller/UserSelfController.java
package com.admin.user.controller;

import com.admin.common.model.dto.MyCodeToKeyDTO;
import com.admin.common.model.vo.ApiResultVO;
import com.admin.user.model.dto.UserSelfDeleteDTO;
import com.admin.user.model.dto.UserSelfUpdateBaseInfoDTO;
import com.admin.user.model.dto.UserSelfUpdateEmailDTO;
import com.admin.user.model.dto.UserSelfUpdatePasswordDTO;
import com.admin.user.model.vo.UserSelfBaseInfoVO;
import com.admin.user.service.UserSelfService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.annotation.Resource;
import javax.validation.Valid;

@RestController
@RequestMapping(value = "/userSelf")
@Api(tags = "用户-自我-管理")
public class UserSelfController {

    @Resource
    UserSelfService baseService;

    @ApiOperation(value = "当前用户：退出登录")
    @PostMapping(value = "/logout")
    public ApiResultVO<String> userSelfLogout() {
        return ApiResultVO.ok(baseService.userSelfLogout());
    }

    @ApiOperation(value = "获取：当前用户，基本信息")
    @PostMapping(value = "/baseInfo")
    public ApiResultVO<UserSelfBaseInfoVO> userSelfBaseInfo() {
        return ApiResultVO.ok(baseService.userSelfBaseInfo());
    }

    @ApiOperation(value = "当前用户：基本信息：修改")
    @PostMapping(value = "/updateBaseInfo")
    public ApiResultVO<String> userSelfUpdateBaseInfo(@RequestBody @Valid UserSelfUpdateBaseInfoDTO dto) {
        return ApiResultVO.ok(baseService.userSelfUpdateBaseInfo(dto));
    }

    @ApiOperation(value = "当前用户：修改密码")
    @PostMapping(value = "/updatePassword")
    public ApiResultVO<String> userSelfUpdatePassword(@RequestBody @Valid UserSelfUpdatePasswordDTO dto) {
        return ApiResultVO.ok(baseService.userSelfUpdatePassword(dto));
    }

    @ApiOperation(value = "当前用户：修改密码，发送，邮箱验证码")
    @PostMapping(value = "/updatePassword/sendEmailCode")
    public ApiResultVO<String> userSelfUpdatePasswordSendEmailCode() {
        return ApiResultVO.ok(baseService.userSelfUpdatePasswordSendEmailCode());
    }

    @ApiOperation(value = "当前用户：修改邮箱")
    @PostMapping(value = "/updateEmail")
    public ApiResultVO<String> userSelfUpdateEmail(@RequestBody @Valid UserSelfUpdateEmailDTO dto) {
        return ApiResultVO.ok(baseService.userSelfUpdateEmail(dto));
    }

    @ApiOperation(value = "当前用户：修改邮箱，发送，邮箱验证码")
    @PostMapping(value = "/updateEmail/sendEmailCode")
    public ApiResultVO<String> userSelfUpdateEmailSendEmailCode() {
        return ApiResultVO.ok(baseService.userSelfUpdateEmailSendEmailCode());
    }

    @ApiOperation(value = "当前用户：修改邮箱，发送，邮箱验证码，验证码兑换 key")
    @PostMapping(value = "/updateEmail/sendEmailCode/codeToKey")
    public ApiResultVO<String> userSelfUpdateEmailSendEmailCodeCodeToKey(@RequestBody @Valid MyCodeToKeyDTO dto) {
        return ApiResultVO.ok(baseService.userSelfUpdateEmailSendEmailCodeCodeToKey(dto));
    }

    @ApiOperation(value = "当前用户：刷新jwt私钥后缀")
    @PostMapping(value = "/refreshJwtSecretSuf")
    public ApiResultVO<String> userSelfRefreshJwtSecretSuf() {
        return ApiResultVO.ok(baseService.userSelfRefreshJwtSecretSuf());
    }

    @ApiOperation(value = "当前用户：注销")
    @PostMapping(value = "/delete")
    public ApiResultVO<String> userSelfDelete(UserSelfDeleteDTO dto) {
        return ApiResultVO.ok(baseService.userSelfDelete(dto));
    }

    @ApiOperation(value = "当前用户：注销，发送，邮箱验证码")
    @PostMapping(value = "/delete/sendEmailCode")
    public ApiResultVO<String> userSelfDeleteSendEmailCode() {
        return ApiResultVO.ok(baseService.userSelfDeleteSendEmailCode());
    }

}

File: ant-react/src/api/UserSelfController.ts
import MyCodeToKeyDTO from "@/model/dto/MyCodeToKeyDTO";
import {AxiosRequestConfig} from "axios";
import $http from "../../util/HttpUtil";

export interface UserSelfBaseInfoVO {
    avatarUrl?: string // 头像url
    bio?: string // 个人简介
    email?: string // 邮箱，会脱敏
    nickname?: string // 昵称
    passwordFlag?: boolean // 是否有密码，用于前端显示，修改密码/设置密码
}

// 用户-自我-管理 获取：当前用户，基本信息
export function userSelfBaseInfo(config?: AxiosRequestConfig) {
    return $http.myPost<UserSelfBaseInfoVO>('/userSelf/baseInfo', undefined, config)
}

// 用户-自我-管理 当前用户：注销
export function userSelfDelete(form: FormData, config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/delete', form, config)
}

// 用户-自我-管理 当前用户：注销，发送，邮箱验证码
export function userSelfDeleteSendEmailCode(config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/delete/sendEmailCode', undefined, config)
}

// 用户-自我-管理 当前用户：退出登录
export function userSelfLogout(config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/logout', undefined, config)
}

// 用户-自我-管理 当前用户：刷新jwt私钥后缀
export function userSelfRefreshJwtSecretSuf(config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/refreshJwtSecretSuf', undefined, config)
}

export interface UserSelfUpdateBaseInfoDTO {
    avatarUrl?: string // 头像url
    bio?: string // 个人简介
    nickname?: string // 昵称
}

// 用户-自我-管理 当前用户：基本信息：修改
export function userSelfUpdateBaseInfo(form: UserSelfUpdateBaseInfoDTO, config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/updateBaseInfo', form, config)
}

export interface UserSelfUpdateEmailDTO {
    code?: string // 新邮箱，验证码
    email?: string // 邮箱
    key?: string // 旧邮箱，验证码兑换的 key
}

// 用户-自我-管理 当前用户：修改邮箱
export function userSelfUpdateEmail(form: UserSelfUpdateEmailDTO, config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/updateEmail', form, config)
}

// 用户-自我-管理 当前用户：修改邮箱，发送，邮箱验证码
export function userSelfUpdateEmailSendEmailCode(config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/updateEmail/sendEmailCode', undefined, config)
}

// 用户-自我-管理 当前用户：修改邮箱，发送，邮箱验证码，验证码兑换 key
export function userSelfUpdateEmailSendEmailCodeCodeToKey(form: MyCodeToKeyDTO, config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/updateEmail/sendEmailCode/codeToKey', form, config)
}

export interface UserSelfUpdatePasswordDTO {
    code?: string // 邮箱验证码
    newOrigPassword?: string // 前端加密之后的原始密码，新密码
    newPassword?: string // 前端加密之后的，新密码
}

// 用户-自我-管理 当前用户：修改密码
export function userSelfUpdatePassword(form: UserSelfUpdatePasswordDTO, config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/updatePassword', form, config)
}

// 用户-自我-管理 当前用户：修改密码，发送，邮箱验证码
export function userSelfUpdatePasswordSendEmailCode(config?: AxiosRequestConfig) {
    return $http.myPost<string>('/userSelf/updatePassword/sendEmailCode', undefined, config)
}
